diff --git a/dist/types.d.ts b/dist/types.d.ts
index dfdf693d1c8c4d13bb78aade9ead76c450175f35..a7112baf27854275dae6d8b6cc5995124adca96e 100644
--- a/dist/types.d.ts
+++ b/dist/types.d.ts
@@ -361,17 +361,20 @@ export interface TxHistoryItem {
         amount: number;
         from_addr: string;
         token_id: string;
+        price?: number;
     }[];
     sends: {
         amount: number;
         to_addr: string;
         token_id: string;
+        price?: number;
     }[];
     time_at: number;
     token_approve: {
         spender: string;
         token_id: string;
         value: number;
+        price?: number;
     } | null;
     tx: {
         eth_gas_fee: number;
@@ -1212,6 +1215,7 @@ export interface SupportedChain {
     is_disabled: boolean;
     explorer_host: string;
     block_interval: number;
+    severity?: number;
 }
 export interface ChainListItem {
     chain_id: number;
