diff --git a/dist/types.d.ts b/dist/types.d.ts
index 32d5eb946090e0ee445891cbb075e1e1e00fde60..96a5346b4014bea8203c0b7352bd6b9e075f96cf 100644
--- a/dist/types.d.ts
+++ b/dist/types.d.ts
@@ -361,17 +361,20 @@ export interface TxHistoryItem {
         amount: number;
         from_addr: string;
         token_id: string;
+        price?: number;
     }[];
     sends: {
         amount: number;
         to_addr: string;
         token_id: string;
+        price?: number;
     }[];
     time_at: number;
     token_approve: {
         spender: string;
         token_id: string;
         value: number;
+        price?: number;
     } | null;
     tx: {
         eth_gas_fee: number;
@@ -1211,6 +1214,8 @@ export interface SupportedChain {
     eip_1559: boolean;
     is_disabled: boolean;
     explorer_host: string;
+    severity?: number;
+    block_interval?: number;
 }
 export interface ChainListItem {
     chain_id: number;
