diff --git a/android/src/main/java/com/reactnativemenu/MenuView.kt b/android/src/main/java/com/reactnativemenu/MenuView.kt
index c45f5cc..b391343 100644
--- a/android/src/main/java/com/reactnativemenu/MenuView.kt
+++ b/android/src/main/java/com/reactnativemenu/MenuView.kt
@@ -9,7 +9,7 @@ import android.text.Spannable
 import android.text.SpannableStringBuilder
 import android.text.style.ForegroundColorSpan
 import android.view.*
-import android.widget.PopupMenu
+import androidx.appcompat.widget.PopupMenu
 import com.facebook.react.bridge.*
 import com.facebook.react.uimanager.UIManagerHelper
 import com.facebook.react.views.view.ReactViewGroup
@@ -19,9 +19,11 @@ import java.lang.reflect.Field
 class MenuView(private val mContext: ReactContext) : ReactViewGroup(mContext) {
   private lateinit var mActions: ReadableArray
   private var mIsAnchoredToRight = false
-  private val mPopupMenu: PopupMenu = PopupMenu(context, this)
+  private var mThemeVariant = "light"
+  private var mPopupMenu: PopupMenu = createPopupMenu()
   private var mIsMenuDisplayed = false
   private var mIsOnLongPress = false
+  private var mIsSuppressNativeLongPress = false
   private var mGestureDetector: GestureDetector
   private var mHitSlopRect: Rect? = null
     set(value) {
@@ -33,7 +35,7 @@ class MenuView(private val mContext: ReactContext) : ReactViewGroup(mContext) {
   init {
     mGestureDetector = GestureDetector(mContext, object : GestureDetector.SimpleOnGestureListener() {
       override fun onLongPress(e: MotionEvent) {
-        if (!mIsOnLongPress) {
+        if (mIsSuppressNativeLongPress || !mIsOnLongPress) {
           return
         }
         prepareMenu()
@@ -89,6 +91,20 @@ class MenuView(private val mContext: ReactContext) : ReactViewGroup(mContext) {
     mIsAnchoredToRight = isAnchoredToRight
   }
 
+  fun setThemeVariant(themeVariant: String?) {
+    val nextThemeVariant = if (themeVariant == "dark") "dark" else "light"
+    if (mThemeVariant == nextThemeVariant) {
+      return
+    }
+
+    mThemeVariant = nextThemeVariant
+    mPopupMenu = createPopupMenu()
+  }
+
+  fun setSuppressNativeLongPress(disabled: Boolean) {
+    mIsSuppressNativeLongPress = disabled
+  }
+
   fun setIsOpenOnLongPress(isLongPress: Boolean) {
     mIsOnLongPress = isLongPress
   }
@@ -96,6 +112,14 @@ class MenuView(private val mContext: ReactContext) : ReactViewGroup(mContext) {
   private val getActionsCount: Int
     get() = mActions.size()
 
+  private fun createPopupMenu(): PopupMenu {
+    val themeResId =
+      if (mThemeVariant == "dark") androidx.appcompat.R.style.Theme_AppCompat
+      else androidx.appcompat.R.style.Theme_AppCompat_Light
+
+    return PopupMenu(ContextThemeWrapper(context, themeResId), this)
+  }
+
   private fun prepareMenuItem(menuItem: MenuItem, config: ReadableMap?) {
     val titleColor = when (config != null && config.hasKey("titleColor") && !config.isNull("titleColor")) {
       true -> config.getInt("titleColor")
diff --git a/android/src/main/java/com/reactnativemenu/MenuViewManagerBase.kt b/android/src/main/java/com/reactnativemenu/MenuViewManagerBase.kt
index e4d2743..1362c2f 100644
--- a/android/src/main/java/com/reactnativemenu/MenuViewManagerBase.kt
+++ b/android/src/main/java/com/reactnativemenu/MenuViewManagerBase.kt
@@ -30,6 +30,16 @@ abstract class MenuViewManagerBase : ReactClippingViewManager<MenuView>() {
     view.setIsAnchoredToRight(isAnchoredToRight)
   }
 
+  @ReactProp(name = "themeVariant")
+  fun setThemeVariant(view: MenuView, themeVariant: String?) {
+    view.setThemeVariant(themeVariant)
+  }
+
+  @ReactProp(name = "suppressNativeLongPress", defaultBoolean = false)
+  fun setSuppressNativeLongPress(view: MenuView, disabled: Boolean) {
+    view.setSuppressNativeLongPress(disabled)
+  }
+
   @ReactProp(name = "shouldOpenOnLongPress", defaultBoolean = false)
   fun setIsOnLongPress(view: MenuView, isOnLongPress: Boolean) {
     view.setIsOpenOnLongPress(isOnLongPress)
diff --git a/lib/commonjs/NativeModuleSpecs/UIMenuNativeComponent.ts b/lib/commonjs/NativeModuleSpecs/UIMenuNativeComponent.ts
index e650935..099af0e 100644
--- a/lib/commonjs/NativeModuleSpecs/UIMenuNativeComponent.ts
+++ b/lib/commonjs/NativeModuleSpecs/UIMenuNativeComponent.ts
@@ -51,7 +51,9 @@ export interface NativeProps extends ViewProps {
 	actions: Array<MenuAction>;
 	actionsHash: string; // just a workaround to make sure we don't have to manually compare MenuActions manually in C++ (since it's a struct and that's a pain)
 	title?: string;
+	isAnchoredToRight?: boolean;
 	themeVariant?: string;
+	suppressNativeLongPress?: boolean;
 	shouldOpenOnLongPress?: boolean;
 	hitSlop: {
 		top: Int32;
diff --git a/lib/commonjs/UIMenuView.android.js b/lib/commonjs/UIMenuView.android.js
index aed934a..e4420d9 100644
--- a/lib/commonjs/UIMenuView.android.js
+++ b/lib/commonjs/UIMenuView.android.js
@@ -8,7 +8,10 @@ var _reactNative = require("react-native");
 var _react = require("react");
 var _jsxRuntime = require("react/jsx-runtime");
 const NativeMenuComponent = (0, _reactNative.requireNativeComponent)("MenuView");
-const MenuComponent = /*#__PURE__*/(0, _react.forwardRef)((props, ref) => {
+const MenuComponent = /*#__PURE__*/(0, _react.forwardRef)(({
+  androidSuppressNativeLongPress = false,
+  ...props
+}, ref) => {
   const nativeRef = (0, _react.useRef)(null);
   (0, _react.useImperativeHandle)(ref, () => ({
     show: () => {
@@ -20,6 +23,7 @@ const MenuComponent = /*#__PURE__*/(0, _react.forwardRef)((props, ref) => {
     }
   }), []);
   return /*#__PURE__*/(0, _jsxRuntime.jsx)(NativeMenuComponent, {
+    suppressNativeLongPress: androidSuppressNativeLongPress,
     ...props,
     ref: nativeRef
   });
diff --git a/lib/module/NativeModuleSpecs/UIMenuNativeComponent.ts b/lib/module/NativeModuleSpecs/UIMenuNativeComponent.ts
index e650935..099af0e 100644
--- a/lib/module/NativeModuleSpecs/UIMenuNativeComponent.ts
+++ b/lib/module/NativeModuleSpecs/UIMenuNativeComponent.ts
@@ -51,7 +51,9 @@ export interface NativeProps extends ViewProps {
 	actions: Array<MenuAction>;
 	actionsHash: string; // just a workaround to make sure we don't have to manually compare MenuActions manually in C++ (since it's a struct and that's a pain)
 	title?: string;
+	isAnchoredToRight?: boolean;
 	themeVariant?: string;
+	suppressNativeLongPress?: boolean;
 	shouldOpenOnLongPress?: boolean;
 	hitSlop: {
 		top: Int32;
diff --git a/lib/module/UIMenuView.android.js b/lib/module/UIMenuView.android.js
index e698dd3..1695eb1 100644
--- a/lib/module/UIMenuView.android.js
+++ b/lib/module/UIMenuView.android.js
@@ -4,7 +4,10 @@ import { findNodeHandle, requireNativeComponent, UIManager } from "react-native"
 import { forwardRef, useImperativeHandle, useRef } from "react";
 import { jsx as _jsx } from "react/jsx-runtime";
 const NativeMenuComponent = requireNativeComponent("MenuView");
-const MenuComponent = /*#__PURE__*/forwardRef((props, ref) => {
+const MenuComponent = /*#__PURE__*/forwardRef(({
+  androidSuppressNativeLongPress = false,
+  ...props
+}, ref) => {
   const nativeRef = useRef(null);
   useImperativeHandle(ref, () => ({
     show: () => {
@@ -16,6 +19,7 @@ const MenuComponent = /*#__PURE__*/forwardRef((props, ref) => {
     }
   }), []);
   return /*#__PURE__*/_jsx(NativeMenuComponent, {
+    suppressNativeLongPress: androidSuppressNativeLongPress,
     ...props,
     ref: nativeRef
   });
diff --git a/lib/typescript/src/NativeModuleSpecs/UIMenuNativeComponent.d.ts b/lib/typescript/src/NativeModuleSpecs/UIMenuNativeComponent.d.ts
index 24ee7fd..7a412d8 100644
--- a/lib/typescript/src/NativeModuleSpecs/UIMenuNativeComponent.d.ts
+++ b/lib/typescript/src/NativeModuleSpecs/UIMenuNativeComponent.d.ts
@@ -44,7 +44,9 @@ export interface NativeProps extends ViewProps {
     actions: Array<MenuAction>;
     actionsHash: string;
     title?: string;
+    isAnchoredToRight?: boolean;
     themeVariant?: string;
+    suppressNativeLongPress?: boolean;
     shouldOpenOnLongPress?: boolean;
     hitSlop: {
         top: Int32;
diff --git a/lib/typescript/src/index.d.ts b/lib/typescript/src/index.d.ts
index 61d210a..5863b75 100644
--- a/lib/typescript/src/index.d.ts
+++ b/lib/typescript/src/index.d.ts
@@ -9,6 +9,7 @@ declare const MenuView: import("react").ForwardRefExoticComponent<{
     isAnchoredToRight?: boolean;
     shouldOpenOnLongPress?: boolean;
     themeVariant?: string;
+    androidSuppressNativeLongPress?: boolean;
     hitSlop?: {
         top: number;
         bottom: number;
diff --git a/lib/typescript/src/types.d.ts b/lib/typescript/src/types.d.ts
index 532ce80..1d8f599 100644
--- a/lib/typescript/src/types.d.ts
+++ b/lib/typescript/src/types.d.ts
@@ -176,4 +176,9 @@ export type NativeMenuComponentProps = {
     hitSlop?: MenuComponentProps["hitSlop"];
     testID?: string;
     ref?: React.ForwardedRef<MenuComponentRef>;
+    isAnchoredToRight?: MenuComponentProps["isAnchoredToRight"];
+    themeVariant?: MenuComponentProps["themeVariant"];
+    suppressNativeLongPress?: boolean;
+    shouldOpenOnLongPress?: MenuComponentProps["shouldOpenOnLongPress"];
+    androidSuppressNativeLongPress?: boolean;
 };
diff --git a/src/NativeModuleSpecs/UIMenuNativeComponent.ts b/src/NativeModuleSpecs/UIMenuNativeComponent.ts
index e650935..099af0e 100644
--- a/src/NativeModuleSpecs/UIMenuNativeComponent.ts
+++ b/src/NativeModuleSpecs/UIMenuNativeComponent.ts
@@ -51,7 +51,9 @@ export interface NativeProps extends ViewProps {
 	actions: Array<MenuAction>;
 	actionsHash: string; // just a workaround to make sure we don't have to manually compare MenuActions manually in C++ (since it's a struct and that's a pain)
 	title?: string;
+	isAnchoredToRight?: boolean;
 	themeVariant?: string;
+	suppressNativeLongPress?: boolean;
 	shouldOpenOnLongPress?: boolean;
 	hitSlop: {
 		top: Int32;
diff --git a/src/UIMenuView.android.tsx b/src/UIMenuView.android.tsx
index 967e375..eb52ea1 100644
--- a/src/UIMenuView.android.tsx
+++ b/src/UIMenuView.android.tsx
@@ -12,7 +12,7 @@ const NativeMenuComponent = requireNativeComponent(
 ) as HostComponent<NativeMenuComponentProps>;
 
 const MenuComponent = forwardRef<MenuComponentRef, NativeMenuComponentProps>(
-	(props, ref) => {
+	({ androidSuppressNativeLongPress = false, ...props }, ref) => {
 		const nativeRef = useRef(null);
 
 		useImperativeHandle(
@@ -31,7 +31,7 @@ const MenuComponent = forwardRef<MenuComponentRef, NativeMenuComponentProps>(
 			[],
 		);
 
-		return <NativeMenuComponent {...props} ref={nativeRef} />;
+		return <NativeMenuComponent suppressNativeLongPress={androidSuppressNativeLongPress} {...props} ref={nativeRef} />;
 	},
 );
 
diff --git a/src/types.ts b/src/types.ts
index 9c323d9..1e76699 100644
--- a/src/types.ts
+++ b/src/types.ts
@@ -169,7 +169,9 @@ type MenuComponentPropsBase = {
 };
 
 export type MenuComponentProps =
-	React.PropsWithChildren<MenuComponentPropsBase>;
+	React.PropsWithChildren<MenuComponentPropsBase> & {
+		androidSuppressNativeLongPress?: boolean;
+	};
 
 export type MenuComponentRef = {
 	show: () => void;
@@ -195,4 +197,9 @@ export type NativeMenuComponentProps = {
 	hitSlop?: MenuComponentProps["hitSlop"];
 	testID?: string;
 	ref?: React.ForwardedRef<MenuComponentRef>;
+	isAnchoredToRight?: MenuComponentProps["isAnchoredToRight"];
+	themeVariant?: MenuComponentProps["themeVariant"];
+	suppressNativeLongPress?: boolean;
+	shouldOpenOnLongPress?: MenuComponentProps["shouldOpenOnLongPress"];
+	androidSuppressNativeLongPress?: boolean;
 };
