diff --git a/android/build.gradle b/android/build.gradle
index ba37f5e3ff97a142c243d63db243d5843062a8a0..896d056a77141d9e769da316d430b809aa3af6bc 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -27,6 +27,7 @@ def DEFAULT_MIN_SDK_VERSION = 16
 def DEFAULT_TARGET_SDK_VERSION = 31
 
 android {
+  namespace 'com.proyecto26.inappbrowser'
   compileSdkVersion safeExtGet('compileSdkVersion', DEFAULT_COMPILE_SDK_VERSION)
   buildToolsVersion safeExtGet('buildToolsVersion', DEFAULT_BUILD_TOOLS_VERSION)
   defaultConfig {
