diff --git a/node_modules/@rabby-wallet/rabby-api/dist/index.d.ts b/node_modules/@rabby-wallet/rabby-api/dist/index.d.ts
index 2097372..24e22d1 100644
--- a/node_modules/@rabby-wallet/rabby-api/dist/index.d.ts
+++ b/node_modules/@rabby-wallet/rabby-api/dist/index.d.ts
@@ -395,12 +395,33 @@ export declare class OpenApiService {
     }>;
     searchDapp: (params: {
         q: string;
-    }) => Promise<BasicDappInfo[]>;
+    }) => Promise<{
+      dapps: BasicDappInfo[];
+      page: {
+        limit: number,
+        start: number,
+        total: number
+      }
+    }>;
     getDappsInfo: (params: {
         ids: string[];
-    }) => Promise<BasicDappInfo[]>;
+    }) => Promise<{
+      dapps: BasicDappInfo[];
+      page: {
+        limit: number,
+        start: number,
+        total: number
+      }
+    }>;
     getHotDapps: (params?: {
         limit: number;
-    }) => Promise<BasicDappInfo[]>;
+    }) => Promise<{
+      dapps: BasicDappInfo[];
+      page: {
+        limit: number,
+        start: number,
+        total: number
+      }
+    }>;
 }
 export {};
